policies/recipes/artifacts.go

```diff
@@ -69,51 +67,52 @@ func FetchArtifacts(ctx context.Context, artifacts []*osconfigpb.SoftwareRecipe_
 
 func fetchArtifact(ctx context.Context, artifact *osconfigpb.SoftwareRecipe_Artifact, directory string) (string, error) {
 	localPath := filepath.Join(directory, artifact.Id)
-	uri, err := url.Parse(artifact.Uri)
-	if err != nil {
-		return "", fmt.Errorf("Could not parse url %q for artifact %q", artifact.Uri, artifact.Id)
-	}
-
 	var reader io.ReadCloser
-	switch strings.ToLower(uri.Scheme) {
-	case "gs":
-		reader, err = fetchWithGCS(ctx, uri.Host, uri.Path, uri.Fragment)
+	var err error
```

Comment:
Is this needed at this scope? It's usually better to scope variables as narrowly as possible.